xarray/conventions.py

```diff
@@ -661,6 +681,14 @@ def _encode_coordinates(variables, attributes, non_dim_coord_names):
             ):
                 variable_coordinates[k].add(coord_name)
 
+            encoding_val = v.encoding.get
+            if (
+                encoding_val("bounds") == coord_name
```

Comment:
nit: Mild preference for writing out `v.encoding.get("bounds")`

Reply:
Done.